code/game/machinery/cloning.dm

```diff
@@ -109,7 +111,27 @@ GLOBAL_VAR_INIT(clones, 0)
 	. = ..()
 	. += "The write-protect tab is set to [read_only ? "protected" : "unprotected"]."
 
+// Biomass
 
+/obj/machinery/clonepod/attackby(obj/item/W, mob/user, params)
+	var/old_biomass = biomass // saves biomas variable
+	if(istype(W, /obj/item))
+		if(istype(W, /obj/item/stack/sheet/animalhide/human)) // lets play flesh or meat
+			biomass += 50
+		if(istype(W, /obj/item/reagent_containers/food/snacks/meat/slab)) // Are we inserting meat?
+			if(istype(W, /obj/item/reagent_containers/food/snacks/meat/slab/human)) // human meat is the easiest to turn into new human materials
+				biomass += 50
+			else if(istype(W, /obj/item/reagent_containers/food/snacks/meat/slab/synthmeat))
+				biomass += 34 // synthmeat can be many different things, thus it should be decently high. This ensures 3 of them gives you a full clone, without fucking with decimals.
+			else if(istype(W, /obj/item/reagent_containers/food/snacks/meat/slab/monkey))
+				biomass += 25 // Monkey meat is close to human, but not actually human.
+			// if(istype W, /obj/item/reagent_containers/) //  this space will eventually be for my biomass cartidge, which aren't done.
+			else
+				biomass = biomass + 20 // Not actually human meat? Means that you need more of it.
+	if(biomass != old_biomass) // deletes the item you inserted if biomass changed.
+		qdel(W)
+	if(biomass > 100)
+		biomass = 100 // no going over 100 biomass, you can't just ingore meat requirements for the rest of the shift because some chemist made 80,000 synthmeat
```

Comment:
Duplicate that check at the top and give a message to the user if theyre trying to feed more meat to an already filled cloner to prevent the chemist from feeding the cloner 50 million synthmeat just for them to be destroyed